File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/metadata/RelationshipMetadata.java
package com.foundgine.core.semantic.metadata;
import com.foundgine.core.abstractions.EntityId;
import com.foundgine.core.abstractions.RelationshipId;
import java.util.List;
public record RelationshipMetadata(RelationshipId id, EntityId source, EntityId target, String name,
                                   ColumnReference sourceKey, ColumnReference targetKey,
                                   boolean collection, List<AliasDeclaration> aliases) {
    public RelationshipMetadata(RelationshipId id, EntityId source, EntityId target, String name,
                                ColumnReference sourceKey, ColumnReference targetKey) {
        this(id,source,target,name,sourceKey,targetKey,true,null);
    }
}

File: src/csharp/Foundgine.Core/Semantic/Metadata/RelationshipMetadata.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic.Metadata;

/// <summary>
/// Static domain relationship metadata. The key mapping identifies how the
/// related domain entities correlate; provider-specific join behavior is
/// deliberately not part of the semantic relationship contract.
/// </summary>
public sealed record RelationshipMetadata(
    RelationshipId Id,
    EntityId Source,
    EntityId Target,
    string Name,
    ColumnReference SourceKey,
    ColumnReference TargetKey,
    bool IsCollection = true,
    IReadOnlyList<AliasDeclaration>? Aliases = null);